hypothesis-python/src/hypothesis/extra/_patching.py

```diff
@@ -141,20 +141,38 @@ def leave_FunctionDef(self, _, updated_node):
 
 
 def get_patch_for(func, failing_examples, *, strip_via=()):
-    # Skip this if we're unable to find the location or source of this function.
+    # Skip this if we're unable to find the location of this function.
     try:
         module = sys.modules[func.__module__]
-        fname = Path(module.__file__).relative_to(Path.cwd())
-        before = inspect.getsource(func)
+        file_path = Path(module.__file__)
     except Exception:
         return None
 
+    fname = (
+        file_path.relative_to(Path.cwd())
+        if file_path.is_relative_to(Path.cwd())
+        else file_path
+    )
```

Comment:
Can happen in hypofuzz if you start the dashboard in an unrelated directory. We may want to disallow or discourage doing that anyway, but even if we do, falling back to the full patch path seems preferable for hypothesis in general rather than erroring and producing no patch.